inference-engine/src/mkldnn_plugin/memory_desc/dnnl_blocked_memory_desc.cpp

```diff
@@ -794,3 +788,41 @@ void DnnlBlockedMemoryDesc::initStrides() {
 void DnnlBlockedMemoryDesc::initOffsetPadding() {
     offsetPaddingToData = VectorDims(std::begin(desc.data.padded_offsets), std::begin(desc.data.padded_offsets) + getOrder().size());
 }
+
+MemoryDescPtr DnnlBlockedMemoryDesc::cloneWithUndefStridesAndOffset() const {
+    DnnlBlockedMemoryDescPtr newDesc = std::make_shared<DnnlBlockedMemoryDesc>(*this);
+    auto &dnnlBlkDesc = newDesc->desc.data.format_desc.blocking;
+    std::fill(std::begin(dnnlBlkDesc.strides), std::begin(dnnlBlkDesc.strides) + getShape().getRank(), DNNL_RUNTIME_DIM_VAL);
+    std::fill(newDesc->strides.begin(), newDesc->strides.begin() + getShape().getRank(), Shape::UNDEFINED_DIM);
+    newDesc->desc.data.offset0 = DNNL_RUNTIME_DIM_VAL;
+    newDesc->status = descStatus::Undefined;
+    return newDesc;
+}
+
+MemoryDescPtr DnnlBlockedMemoryDesc::cloneWithDefaultStridesAndOffset() const {
+    DnnlBlockedMemoryDescPtr newDesc = std::make_shared<DnnlBlockedMemoryDesc>(*this);
+    const auto &rank = getShape().getRank();
+    auto &newDescStrides = newDesc->strides;
+    auto &dnnlBlkDesc = newDesc->desc.data.format_desc.blocking;
+    if (std::any_of(blockedDims.begin(), blockedDims.end(), [](Dim val) { return val == DNNL_RUNTIME_DIM_VAL; })) {
+        std::fill(std::begin(dnnlBlkDesc.strides), std::begin(dnnlBlkDesc.strides) + rank, DNNL_RUNTIME_DIM_VAL);
+        std::fill(newDescStrides.begin(), newDescStrides.begin() + rank, Shape::UNDEFINED_DIM);
+    } else {
+        newDescStrides[order.size() - 1] = 1;
+        for (size_t i = 2; i <= order.size(); i++) {
+            newDescStrides[order.size() - i] = newDescStrides[order.size() - (i - 1)] * blockedDims[blockedDims.size() - (i - 1)];
+        }
+        for (size_t i = 0; i < rank; i++) {
+            dnnlBlkDesc.strides[order[i]] = newDescStrides[i];
+        }
```

Comment:
looks like a code duplication with `DnnlBlockedMemoryDesc(InferenceEngine::Precision prc, const Shape& shape, const VectorDims& blockedDims, const VectorDims& order, size_t offsetPadding, const VectorDims& offsetPaddingToData, const VectorDims& strides) ` constructor strides calculation part. May be it makes sense to extract the common code into some subroutine?

Reply:
there is difference between filling oneDnn desc strides into ctor:
`dnn_blk_desc.strides[i] = tmpStrides[outer_order[i]];`
and into `cloneWithDefaultStridesAndOffset`:
`dnnlBlkDesc.strides[order[i]] = newDescStrides[i];`

also I am not sure that we really need extract strides calculation into separate method (3 lines of code)

Reply:
why do we have such difference? It looks like these algorithms should be the same. As for the subroutine extraction, 3 lines of code is pretty small code frame, but whet it is sprinkled across the code, it may be a problem.

Reply:
strides computation aligned and extracted into separate method